File: AITutor/AI_Tutor/AI_Tutor/Models/ExpertSolution.swift
// AI_Tutor/Models/ExpertSolution.swift
import Foundation
import Observation

enum ExpertType: Equatable, Hashable {
    case a, b, c

    var displayName: String {
        switch self {
        case .a: return "解法一"
        case .b: return "解法二"
        case .c: return "专家总评"
        }
    }

    var modelName: String {
        switch self {
        case .a, .b: return "google/gemini-3-pro-preview"
        case .c:     return "anthropic/claude-sonnet-4.5"
        }
    }
}

@Observable
class ExpertSolution: Identifiable {
    let id = UUID()
    var expert: ExpertType
    var content: String = ""
    var isStreaming: Bool = false
    var isComplete: Bool = false
    var errorMessage: String? = nil

    init(expert: ExpertType) {
        self.expert = expert
    }
}

File: AITutor/app/src/main/java/com/ai/aitutorandroid/models/ExpertSolution.kt
package com.ai.aitutorandroid.models

import java.util.UUID

enum class ExpertType { A, B, C }

data class ExpertSolution(
    val id: String = UUID.randomUUID().toString(),
    val expert: ExpertType,
    val content: String = "",
    val isStreaming: Boolean = false,
    val isComplete: Boolean = false,
    val errorMessage: String? = null
)
